internal/dto/registry.go

```diff
@@ -61,20 +63,26 @@ func (registries Registries) Parse(logger log.Logger) (Registries, error) {
 		username_envvar := registries[i].Username_Envvar
 		password_envvar := registries[i].Password_Envvar
 		namespace_envvar := registries[i].Namespace_Envvar
+		quay_custom_namespace_envvar := registries[i].Quay_Custom_Namespace_Envvar
 		username := LookupEnvVar(username_envvar, logger).Return_value
 		password := LookupEnvVar(password_envvar, logger).Return_value
 		namespace := LookupEnvVar(namespace_envvar, logger).Return_value
+		quay_custom_namespace := LookupEnvVar(quay_custom_namespace_envvar, logger).Return_value
 		if registries[i].ValidCredentials(username) {
 			registries[i].Username = username
 			registries[i].Password = password
+		} else {
+			logger.Infof("Missing credentials for %s\n", registries[i].Url)
+			misconfigured_registries[strconv.FormatInt(int64(i), 10)] = PlaceHolder
+		}
+		if quay_custom_namespace != "" && registries[i].Url == "quay.io" {
+			registries[i].Namespace = quay_custom_namespace
+		} else {
 			inferred_namespace, err := InferNamespace(namespace, username)
 			if err != nil {
 				return nil, err
 			}
 			registries[i].Namespace = inferred_namespace
-		} else {
-			logger.Infof("Missing credentials for %s\n", registries[i].Url)
-			misconfigured_registries[strconv.FormatInt(int64(i), 10)] = PlaceHolder
 		}
```

Comment:
With this following the credentials validation step in the new code, but not the old code, is it okay for it to set the registry namespace on failed credentials?

Reply:
I did have it originally in the conditional if statements, however golang-lint complained it was too complex for another if statement. I will work on this and see if I can bring it back in, or find another solution as well

Reply:
I am looking into refactoring this whole section where it does a check for valid creds, and if fails continues the range of other registries instead of continuing with code execution of a bad registry

Reply:
The `continue` statement should be enough. It moves on to the next iteration.

Reply:
Updated